frontend/src/hooks/useGeocoderRepository.ts

```diff
@@ -0,0 +1,68 @@
+import { AxiosResponse } from 'axios';
+import { useCallback } from 'react';
+import { toast } from 'react-toastify';
+
+import { useApiGeocoder } from './pims-api/useApiGeocoder';
+import { useApiRequestWrapper } from './pims-api/useApiRequestWrapper';
+import { IGeocoderPidsResponse, IGeocoderResponse } from './useApi';
+
+export const useGeocoderRepository = () => {
+  const { getSitePidsApi, searchAddressApi, getNearestToPointApi } = useApiGeocoder();
+
+  // Retrieve by site id
+  const { execute: executeSitePids, loading: isLoadingSitePids } = useApiRequestWrapper<
+    (siteId: string) => Promise<AxiosResponse<IGeocoderPidsResponse>>
+  >({
+    requestFunction: useCallback(async (siteId: string) => await getSitePidsApi(siteId), [
+      getSitePidsApi,
+    ]),
+    requestName: 'getSitePids',
+    onError: useCallback(axiosError => {
+      toast.error(
+        `Failed to get PIMS property data. error from backend: ${axiosError?.response?.data.error}`,
```

Comment:
I wonder if we should indicate a failure in geocoder here or if that would confuse the users?

Reply:
That is something we will need to check with the BA team. At some point, the users will need to report errors to someone. 
"What is the best way to show those so that the issues can be found quickly?" answering that will give us a clear idea of how to proceed.